modules/builds-configmap-creating-from-directories.adoc

```diff
@@ -0,0 +1,128 @@
+// Module included in the following assemblies:
+//
+//* builds/builds-configmaps.adoc
+
+[id="builds-configmap-creating-from-directories_{context}"]
+= Creating a ConfigMap from a directory
+
+You can create a ConfigMap from a directory.
+
+.Procedure
+
+The following example procedure outlines how to create a ConfigMap from a directory.
+
+. Start with a directory with some files that already contain the data with which you want to populate a ConfigMap:
++
+[source,terminal]
+----
+$ ls example-files
+----
++
+.Example output
+[source,terminal]
+----
+game.properties
+ui.properties
+----
++
+[source,terminal]
+----
+$ cat example-files/game.properties
+----
++
+.Example output
+[source,terminal]
+----
+enemies=aliens
+lives=3
+enemies.cheat=true
+enemies.cheat.level=noGoodRotten
+secret.code.passphrase=UUDDLRLRBABAS
+secret.code.allowed=true
+secret.code.lives=30
+----
++
+[source,terminal]
+----
+$ cat example-files/ui.properties
+----
++
+.Example output
+[source,terminal]
+----
+color.good=purple
+color.bad=yellow
+allow.textmode=true
+how.nice.to.look=fairlyNice
+----
+
+. Create a ConfigMap holding the content of each file in this directory by entering the following command:
++
+[source,terminal]
+----
+$ oc create configmap game-config \
+    --from-file=example-files/
+----
++
+When the `--from-file` option points to a directory, each file directly in that directory is used to populate a key in the ConfigMap, where the name of the key is the file name, and the value of the key is the content of the file.
++
+For example, the previous command creates the following ConfigMap:
++
+[source,terminal]
+----
+$ oc describe configmaps game-config
+----
++
+.Example output
+[source,terminal]
+----
+Name:           game-config
+Namespace:      default
+Labels:         <none>
+Annotations:    <none>
+
+Data
+
+game.properties:        121 bytes
+ui.properties:          83 bytes
+----
++
```

Comment:
When run oc describe configmaps, I see that output for these are shown as below , file sizes are 158 & 83 bytes respectively.
[ramakasturinarra@dhcp35-60 openshift-client-linux-4.4.0-0.nightly-2020-09-13-231918]$ ./oc describe configmaps game-config
Name:         game-config
Namespace:    default
Labels:       <none>
Annotations:  <none>

Data
====
game.properties:
----
enemies=aliens
lives=3
enemies.cheat=true
enemies.cheat.level=noGoodRotten
secret.code.passphrase=UUDDLRLRBABAS
secret.code.allowed=true
secret.code.lives=30

ui.properties:
----
color.good=purple
color.bad=yellow
allow.textmode=true
how.nice.to.look=fairlyNice

Events:  <none>
[ramakasturinarra@dhcp35-60 openshift-client-linux-4.4.0-0.nightly-2020-09-13-231918]$ ls -l /tmp/example-files/
total 8
-rw-rw-r--. 1 ramakasturinarra ramakasturinarra 158 Sep 14 12:40 game.properties
-rw-rw-r--. 1 ramakasturinarra ramakasturinarra  83 Sep 14 12:40 ui.properties

Reply:
Thank you! Updating.